src/v/cluster/tm_stm.cc

```diff
@@ -266,7 +318,7 @@ tm_stm::do_update_tx(tm_transaction tx, model::term_id term) {
         co_return tm_stm::op_status::unknown;
     }
 
-    auto tx_opt = co_await get_tx(tx.id);
+    auto tx_opt = do_get_tx(tx.id);
```

Comment:
slightly confused here. weren't they all converted into coroutines in the previous commit and now they are converted back to normal functions?

Reply:
Oops I missed this comment somehow.. so the series of commits is like this..

1. co_routinze get_tx  (cosmetic change).. this (update_tx) being a caller has to be updated to use the couroutine version..
2. Redo get_tx (to handle `transferring` tx) 

As a part of (2) we call `update_tx` because we have to commit the tx change to the log.. now update_tx can directly lookup in the local state because it guarantees that the state machine has applied the changes... I could've squashed (1) and (2) and avoided this confusion but I wanted to separate the meat of the patch (2) into a separate commit.